### PR TITLE
Update names in floatingSuggestons + expose dependencies for UnifiedPicker

### PR DESCRIPTION
**Pull request by Adjective-Object**
(In implementing UnifiedPickers as split controlled/uncontrolled components, I had to make some edits to FloatingSuggestions. This PR has those edits)

#### Pull request checklist

~~- [ ] Addresses an existing issue: Fixes #0000~~
- [x] Include a change request file using `$ npm run change`
  - I'm not familiar with the semantics of breaking changes in `experiments`. Since this has a rename on an externally facing API technically by semver it's a breaking change?

#### Description of changes

Since these contracts were lifted form FloatingPicker, they carried some imprecisenames over with them, This PR fixes that ambiguity.

In addition, since BaseExtendedPicker depended directly on FloatingPicker (not IFloatingPicker), some needed functionality was not exposed on the ref type.

Rather than lie about the type with a cast, I expose those dependencies here.

*Note that the main reason these hooks are being exposed here is because the first draft of Unified Pickers depends on an uncontrolled FloatingSuggestions (e.g. it allows FloatingSuggestions to maintain its own internal state). That shouldn't be the case going forward once FloatingSuggestions is split, so we should reassess the external api then*

#### Focus areas to test

SelectedPeopleList / SelectedTagList

###### Microsoft Reviewers: [Open in CodeFlow](http://wpcp.azurewebsites.net/CodeFlowProtocolProxy2.php?pullrequest=https://github.com/OfficeDev/office-ui-fabric-react/pull/9675)

### CODE REVIEW

File: packages/experiments/src/components/FloatingSuggestions/FloatingSuggestions.tsx [hunk 1]
```diff
@@ -210,14 +210,14 @@ export class FloatingSuggestions<TItem> extends BaseComponent<IFloatingSuggestio
     ) : null;
   }
 
-  private _onChange(item: TItem): void {
-    if (this.props.onChange) {
-      this.props.onChange(item);
+  private _onInsertItem(item: TItem): void {
```
Comment (joschect) [inline packages/experiments/src/components/FloatingSuggestions/FloatingSuggestions.tsx]: This seems like it should be _onSuggestionSelected but that's a minor nit.

**Comment by KevinTCoughlin** [comment]
@Adjective-Object it looks like there's a CI failure:

```shell
[11:59:42 PM] x Error detected while running 'ts:esm'
[11:59:42 PM] x ------------------------------------
[11:59:42 PM] x Error: Command failed: "/opt/hostedtoolcache/node/10.16.0/x64/bin/node" "/home/vsts/work/1/s/common/temp/node_modules/typescript/lib/tsc.js" --pretty --target es5 --outDir lib --module es2015 --project "/home/vsts/work/1/s/packages/experiments/tsconfig.json"

    at ChildProcess.exithandler (child_process.js:294:12)
    at ChildProcess.emit (events.js:198:13)
    at ChildProcess.EventEmitter.emit (domain.js:466:23)
    at maybeClose (internal/child_process.js:982:16)
    at Process.ChildProcess._handle.onexit (internal/child_process.js:259:5)
[11:59:42 PM] x stdout:
[11:59:42 PM] x ‌src&#x2F;components&#x2F;SelectedItemsList&#x2F;Items&#x2F;subcomponents&#x2F;DefaultEditingItem.tsx‌:‌48‌:‌3‌ - ‌error‌ TS2344: ‌Type &#x27;&quot;onChange&quot; | &quot;componentRef&quot; | &quot;inputElement&quot; | &quot;onRemoveSuggestion&quot; | &quot;onSuggestionsHidden&quot;&#x27; does not satisfy the constraint &#x27;&quot;className&quot; | &quot;componentRef&quot; | &quot;suggestionsStore&quot; | &quot;onZeroQuerySuggestion&quot; | &quot;inputElement&quot; | &quot;onRenderSuggestionsItem&quot; | &quot;onResolveSuggestions&quot; | &quot;onInputChanged&quot; | &quot;resolveDelay&quot; | ... 9 more ... | &quot;suggestionItems&quot;&#x27;.‌
  Type '"onChange"' is not assignable to type '"className" | "componentRef" | "suggestionsStore" | "onZeroQuerySuggestion" | "inputElement" | "onRenderSuggestionsItem" | "onResolveSuggestions" | "onInputChanged" | "resolveDelay" | ... 9 more ... | "suggestionItems"'.
```

https://uifabric.visualstudio.com/fabricpublic/_build/results?buildId=19768&view=logs&jobId=12f1170f-54f2-53f3-20dd-22fc7dff55f9&taskId=bd05475d-acb5-5619-3ccb-c46842dbc997&lineStart=2465&lineEnd=2478&colStart=1&colEnd=2

**Comment by Adjective-Object** [comment]
I'm getting failures on snapshots, but when I `npm run update-snapshots` locally, I don't get any diff :|

**Comment by Adjective-Object** [comment]
I don't see how these snapshot changes come out of my changes here, but it looks like Pivot's snapshots update #f3f2f1 -> #ffffff when I run update-snapshots in packages/office-ui-fabric-react.

image

I'm pushing a new commit to this branch with results for referenc,e @KevinTCoughlin  Do you have any idea what's going on here?

**Comment by jdhuntington** [comment]
@Adjective-Object I believe a snapshot change was merged within the last few hours to fix the pivot issue. If you rebase, I believe you'll find that the snapshot changes to pivot can fall off.

**Pull request merged**